crates/edr_eth/src/remote/cacheable_method_invocation.rs

```diff
@@ -86,193 +72,127 @@ enum CacheableMethodInvocation<'a> {
+            RequestMethod::GetCode(address, block_spec) => Ok(CacheableRequestMethod::GetCode {
+                address,
+                block_spec: block_spec.try_into()?,
+            }),
+            RequestMethod::GetLogs(params) => Ok(CacheableRequestMethod::GetLogs {
                 params: params.try_into()?,
             }),
-            MethodInvocation::GetStorageAt(address, position, block_spec) => {
-                Ok(CacheableMethodInvocation::GetStorageAt {
+            RequestMethod::GetStorageAt(address, position, block_spec) => {
+                Ok(CacheableRequestMethod::GetStorageAt {
                     address,
                     position,
                     block_spec: block_spec.try_into()?,
                 })
             }
-            MethodInvocation::GetTransactionByBlockHashAndIndex(block_hash, index) => Ok(
-                CacheableMethodInvocation::GetTransactionByBlockHashAndIndex { block_hash, index },
-            ),
-            MethodInvocation::GetTransactionByBlockNumberAndIndex(block_spec, index) => Ok(
-                CacheableMethodInvocation::GetTransactionByBlockNumberAndIndex {
-                    block_spec: block_spec.try_into()?,
-                    index,
-                },
-            ),
-            MethodInvocation::GetTransactionByHash(transaction_hash) => {
-                Ok(CacheableMethodInvocation::GetTransactionByHash { transaction_hash })
+            RequestMethod::GetTransactionByHash(transaction_hash) => {
+                Ok(CacheableRequestMethod::GetTransactionByHash { transaction_hash })
             }
-            MethodInvocation::GetTransactionCount(address, block_spec) => {
-                Ok(CacheableMethodInvocation::GetTransactionCount {
+            RequestMethod::GetTransactionCount(address, block_spec) => {
+                Ok(CacheableRequestMethod::GetTransactionCount {
                     address,
                     block_spec: block_spec.try_into()?,
                 })
             }
-            MethodInvocation::GetTransactionReceipt(transaction_hash) => {
-                Ok(CacheableMethodInvocation::GetTransactionReceipt { transaction_hash })
+            RequestMethod::GetTransactionReceipt(transaction_hash) => {
+                Ok(CacheableRequestMethod::GetTransactionReceipt { transaction_hash })
             }
-            MethodInvocation::NetVersion(_) => Ok(CacheableMethodInvocation::NetVersion),
+            RequestMethod::NetVersion(_) => Ok(CacheableRequestMethod::NetVersion),
 
             // Explicit to make sure if a new method is added, it is not forgotten here.
-            MethodInvocation::Accounts(_)
-            | MethodInvocation::BlockNumber(_)
-            | MethodInvocation::Call(_, _, _)
-            | MethodInvocation::Coinbase(_)
-            | MethodInvocation::EstimateGas(_, _)
-            | MethodInvocation::FeeHistory(_, _, _)
```

Comment:
We need to keep the `FeeHistory` for the implementation of `eth_FeeHistory`. See `node.ts` in Hardhat

Reply:
Thanks, fixed in: [8c20657](https://github.com/NomicFoundation/hardhat/pull/4687/commits/8c20657c1125d4f3efba6b012f7b1d9c9d2889f3)